src/meta/src/hummock/manager/compaction/mod.rs

```diff
@@ -728,6 +729,46 @@ impl HummockManager {
                 }
             }
 
+            // Filter out the table that has a primary key prefix watermark.
+            let table_id_with_pk_prefix_watermark: HashSet<_> = self
+                .metadata_manager
+                .catalog_controller
+                .get_table_by_ids(
+                    version
+                        .latest_version()
+                        .table_watermarks
+                        .keys()
+                        .map(|id| id.table_id() as _)
+                        .collect(),
+                )
+                .await
+                .map_err(|e| Error::Internal(e.into()))?
+                .into_iter()
+                .filter_map(|table| {
+                    // pk prefix watermark.
+                    if table.clean_watermark_index_in_pk.is_none()
+                        || table.clean_watermark_index_in_pk.unwrap() == 0
```

Comment:
Does this assume the pk prefix watermark can only be a single column? Do we have a sanity check somewhere to make sure this assumption holds?